server/endpoint/webhook/endpoint.go

```diff
@@ -0,0 +1,106 @@
+package webhook
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	kitendpoint "github.com/go-kit/kit/endpoint"
+	kithttp "github.com/go-kit/kit/transport/http"
+
+	"github.com/giantswarm/auto-oncall/server/middleware"
+	"github.com/giantswarm/auto-oncall/service"
+)
+
+const (
+	// Method is the HTTP method this endpoint is registered for.
+	Method = "POST"
+	// Name identifies the endpoint. It is aligned to the package path.
+	Name = "webhook"
+	// Path is the HTTP request path this endpoint is registered for.
+	Path = "/webhook"
+)
+
+// Config represents the configuration used to create a version endpoint.
+type Config struct {
+	// Dependencies.
+	Logger     micrologger.Logger
+	Middleware *middleware.Middleware
+	Service    *service.Service
+}
+
+// New creates a new configured version endpoint.
+func New(config Config) (*Endpoint, error) {
+	// Dependencies.
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "logger must not be empty")
+	}
+	if config.Service == nil {
+		return nil, microerror.Maskf(invalidConfigError, "service must not be empty")
+	}
+
+	newEndpoint := &Endpoint{
+		Config: config,
+	}
+
+	return newEndpoint, nil
+}
+
+type Endpoint struct {
+	Config
+}
+
+func (e *Endpoint) Decoder() kithttp.DecodeRequestFunc {
+	return func(ctx context.Context, r *http.Request) (interface{}, error) {
+		return r, nil
+	}
+}
+
+func (e *Endpoint) Encoder() kithttp.EncodeResponseFunc {
+	return func(ctx context.Context, w http.ResponseWriter, response interface{}) error {
+		endpointResponse := response.(*Response)
+		w.Header().Set("Content-Type", "application/json; charset=utf-8")
+		w.WriteHeader(endpointResponse.StatusCode)
+		return json.NewEncoder(w).Encode(endpointResponse.Body)
+	}
+}
+
+func (e *Endpoint) Endpoint() kitendpoint.Endpoint {
+	return func(ctx context.Context, request interface{}) (interface{}, error) {
+
+		r := request.(*http.Request)
+
+		response := DefaultResponse()
+
+		h, err := e.Service.Webhook.NewHook(r)
+		if err != nil {
+			response.Body.Message = err.Error()
+			response.StatusCode = http.StatusBadRequest
+		} else {
+			response.Body.Message = "webhook request received"
+			response.StatusCode = http.StatusOK
+		}
+
+		go e.Service.Webhook.Process(h)
```

Comment:
Operator? No. That's would be too much for this. 
Deployment event happens -> github sends webhook to auto-oncall -> app dispatches payload and responds to github it received valid webhook -> async process creates routingrule

Reply:
I don't see why the processing has to be split out of the sync request. I think it is just fine then to remove the goroutine and add some backoff to be a bit more safe.

Reply:
Processing is split of the sync, because the success of creating a routing rule is not a part of the response to github about webhook state. Github cares about the delivery of the webhook with payload, not about internal processing. E.g., github shouldn't get error about webhook delivery is opsgenie is down. At least that's how I see it

Reply:
That makes sense. Then let's go with this and iterate in it. I think for the future it would make sense to create a CR synchronously and have an operator reconcile it accordingly. That way all logic is more or less guaranteed. Right now it is not, will fail once and be forgotten.

Reply:
:+1: 
all other review points have been addressed.